### PR TITLE
Internal java solution for CSES Movie Festival II

### PR DESCRIPTION
**Pull request by nathangong**

### CODE REVIEW

File: solutions/cses-1632.mdx [hunk 1]
```diff
@@ -69,4 +68,126 @@ int main() {
...
+	static class Interval {
+		int start, end;
+
+		Interval(int start, int end) {
+			this.start = start;
+			this.end = end;
+		}
+	}
+
+	// BeginCodeSnip{FastIO}
```
Comment (SansPapyrus683) [inline solutions/cses-1632.mdx:132]: i take it kattio doesn't work?

File: solutions/cses-1632.mdx [hunk 2]
```diff
@@ -69,4 +68,126 @@ int main() {
...
+		int k = io.nextInt();
+
+		Interval[] movies = new Interval[n];
+		for (int i = 0; i < n; i++) {
+			movies[i] = new Interval(io.nextInt(), io.nextInt());
+		}
+		// sort movies based on end time
+		Arrays.sort(movies, Comparator.comparingInt(movie -> movie.end));
+
+		int ans = 0;
```
Comment (SansPapyrus683) [inline solutions/cses-1632.mdx:95]: maybe rename to `maxMovies`

File: solutions/cses-1632.mdx [hunk 3]
```diff
@@ -69,4 +68,126 @@ int main() {
...
+		// times when members will finish watching their movies
+		TreeMap<Integer, Integer> endTimes = new TreeMap<>();
+		endTimes.put(0, k); // initialize all members at time 0
```
Comment (SansPapyrus683) [inline solutions/cses-1632.mdx:98]: put two spaces after a line of code before you start a comment
```suggestion
		endTimes.put(0, k);  // initialize all members at time 0
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/cses-1632.mdx (hunk 1 above)
Comment (nathangong) [reply solutions/cses-1632.mdx:132]: yeah i added a warning block for it

**Review by SansPapyrus683 — APPROVED** [review]
wow didn't think you'd take my `ans` suggestion

**Review by brad182 — APPROVED** [review]
Sanspapyrus683 has nothing more to say so it definitely is good

File: solutions/cses-1632.mdx [hunk 4]
```diff
@@ -2,22 +2,20 @@
...
-assign it to one of the $K$ members to watch (or none of them).
+assign it to one of the $k$ members to watch (or none of them).
 
 Keep track of the time each member finishes watching all of his currently
-assigned movies in a sorted collection (e.g. a
-[`multiset`](https://en.cppreference.com/w/cpp/container/multiset) in C++).
-Initially, the collection consists of $K$ zeroes.
+assigned movies in an ordered multiset
+([represented](https://usaco.guide/silver/intro-ordered/?lang=java#multisets) by
+a `TreeMap` in Java). Initially, the collection consists of $k$ zeroes.
```
Comment (dongliu0426) [inline solutions/cses-1632.mdx:18]:
```suggestion
([represented](/silver/intro-ordered/#multisets) by
a `TreeMap` in Java or a `multiset` in C++). Initially, the collection consists of $k$ zeroes.
```

**Review by dongliu0426 — CHANGES_REQUESTED** [review]

**Comment by dongliu0426** [comment]
(merging if checks pass)

**Pull request merged**